Apps/WebClient/src/Server/Services/WalletService.cs

```diff
@@ -0,0 +1,352 @@
+            RequestResult<PatientModel> patientRequestResult =
+                await this.clientRegistriesDelegate.GetDemographicsByHDIDAsync(hdId).ConfigureAwait(true);
+            if (patientRequestResult.ResultStatus != ResultType.Success)
+            {
+                this.logger.LogDebug($"Error getting patient info {hdId}: {JsonSerializer.Serialize(patientRequestResult)}");
+                return new RequestResult<IEnumerable<WalletCredentialModel>>()
+                {
+                    ResultStatus = ResultType.Error,
+                    ResultError = new RequestResultError()
+                    {
+                        ResultMessage = "Error getting patient info when creating wallet credential",
+                        InnerError = patientRequestResult.ResultError,
+                    },
+                };
+            }
+
+            this.logger.LogDebug($"Finished getting patient info {JsonSerializer.Serialize(patientRequestResult)}");
+            foreach (string targetId in targetIds)
+            {
+                this.logger.LogDebug($"Getting immunization info {targetId}");
+                RequestResult<ImmunizationEvent> immunizationResult =
+                    await this.immunizationDelegate.GetImmunization(hdId, targetId).ConfigureAwait(true);
+                if (immunizationResult.ResultStatus != ResultType.Success)
+                {
+                    this.logger.LogDebug($"Error getting immunization info {targetId}: {JsonSerializer.Serialize(immunizationResult)}");
+                    return new RequestResult<IEnumerable<WalletCredentialModel>>()
+                    {
+                        ResultStatus = ResultType.Error,
+                        ResultError = new RequestResultError()
+                        {
+                            ResultMessage = "Error getting immunization info when creating wallet credential",
+                            InnerError = patientRequestResult.ResultError,
+                        },
+                    };
+                }
+
+                this.logger.LogDebug($"Finished getting immunization info {targetId}: {JsonSerializer.Serialize(immunizationResult)}");
+                ImmunizationEvent immunization = immunizationResult.ResourcePayload!;
+                ImmunizationCredentialPayload credentialPayload = new ImmunizationCredentialPayload()
+                {
+                    Provider = immunization.ProviderOrClinic,
+                    LotNumber = string.Join(",", immunization.Immunization.ImmunizationAgents.Select(immz => immz.LotNumber)),
+                    ImmunizationType = immunization.Immunization.Name,
+                    ImmunizationProduct = string.Join(",", immunization.Immunization.ImmunizationAgents.Select(immz => immz.ProductName)),
+                    ImmunizationAgent = string.Join(",", immunization.Immunization.ImmunizationAgents.Select(immz => immz.Name)),
```

Comment:
This is kinda misleading since there could be possible multiple agents.
I think in the future we should rename the field to ImmunizationAgents and use the concatenation or change the schema to take multiple agent fields since it has to be flat

Reply:
I don't think we have a solution for multiple agents right now, we might have to discuss with the team